src/scaffoldmaker/annotation/annotationgroup.py

```diff
@@ -363,12 +365,36 @@ def setMarkerMaterialCoordinates(self, materialCoordinatesField, materialCoordin
                         return
                     if not isinstance(xi, list):
                         xi = [xi]  # workaround for Zinc 1-D xi being a plain float
+                    markerLocation = getAnnotationMarkerLocationField(fieldmodule, mesh)
+                    markerLocation.assignMeshLocation(fieldcache, element, xi)
+                    self._markerMaterialCoordinatesField.assignReal(fieldcache, materialCoordinates)
                     del findMeshLocation
                     del constCoordinates
                 else:
                     element, xi = self.getMarkerLocation()
-                # use this function to reassign material coordinates to be within mesh
-                self.setMarkerLocation(element, xi)
+                    # following stores material coordinates at the current location:
+                    fieldcache.setMeshLocation(element, xi)
+                    result, materialCoordinates = self._markerMaterialCoordinatesField.evaluateReal(
+                        fieldcache, self._markerMaterialCoordinatesField.getNumberOfComponents())
+                    fieldcache.setNode(markerNode)
+                    self._markerMaterialCoordinatesField.assignReal(fieldcache, materialCoordinates)
+
+    def evaluateMaterialCoordinatesFromElementXi(self, materialCoordinatesField):
+        """
+        Calculate the material coordinates from element and xi values.
+        :param materialCoordinatesField: Material coordinates field
+        :return: material coordinates derived from element and xi
+        """
+        fieldmodule = self._group.getFieldmodule()
```

Comment:
In the function docs say "For a marker annotation group, calculate..."
And before other code in the function add line:
```
assert self._isMarker
```